Update for 1.20 and 1.23 scans (#209)
rancher/rancher-docs: docs/reference-guides/rancher-security/rancher-v2.6-hardening-guides/rke1-self-assessment-guide-with-cis-v1.6-benchmark.md

---
title: RKE CIS v1.6 Benchmark - Self-Assessment Guide - Rancher v2.6
weight: 101
aliases:
  - /rancher/v2.6/en/security/hardening-guides/1.6-benchmark-2.6/
---

### RKE CIS v1.6 Kubernetes Benchmark - Rancher v2.6 with Kubernetes v1.16 to v1.18

[Click here to download a PDF version of this document](https://releases.rancher.com/documents/security/2.6/Rancher_v2-6_CIS_v1-6_Benchmark_Assessment.pdf).

#### Overview

> This document is a companion to the [Rancher v2.6 RKE security hardening guide]({{<baseurl>}}/rancher/v2.6/en/security/hardening-guides/rke-1.6-hardening-2.6/). The hardening guide provides prescriptive guidance for hardening a production installation of Rancher, and this benchmark guide is meant to help you evaluate the level of security of the hardened cluster against each control in the benchmark.

Reviewer: ```suggestion
This document is a companion to the [Rancher v2.6 RKE security hardening guide](./rke1-hardening-guide-with-cis-v1.6-benchmark.md). The hardening guide provides prescriptive guidance for hardening a production installation of Rancher, and this benchmark guide is meant to help you evaluate the level of security of the hardened cluster against each control in the benchmark.

```

```

### 1.2.2 Ensure that the --basic-auth-file argument is not set (Automated)

**Result:** pass

**Remediation:**
Follow the documentation and configure alternate mechanisms for authentication. Then,
edit the API server pod specification file /etc/kubernetes/manifests/kube-apiserver.yaml

> on the master node and remove the --basic-auth-file=<filename> parameter.

Reviewer: ```suggestion
on the master node and remove the `--basic-auth-file=<filename>` parameter.

```

Converting to inline code since Docusaurus will throw an error because it treats `<filename>` as an HTML tag.

Author: resolved

```

### 1.2.3 Ensure that the --token-auth-file parameter is not set (Automated)

**Result:** pass

**Remediation:**
Follow the documentation and configure alternate mechanisms for authentication. Then,
edit the API server pod specification file /etc/kubernetes/manifests/kube-apiserver.yaml

> on the master node and remove the --token-auth-file=<filename> parameter.

Reviewer: ```suggestion
on the master node and remove the `--token-auth-file=<filename>` parameter.

```

Converting to inline code since Docusaurus will throw an error because it treats `<filename>` as an HTML tag.

Author: resolved

```

### 1.2.6 Ensure that the --kubelet-certificate-authority argument is set as appropriate (Automated)

**Result:** pass

**Remediation:**
Follow the Kubernetes documentation and setup the TLS connection between
the apiserver and kubelets. Then, edit the API server pod specification file
/etc/kubernetes/manifests/kube-apiserver.yaml on the master node and set the
--kubelet-certificate-authority parameter to the path to the cert file for the certificate authority.

> --kubelet-certificate-authority=<ca-string>

Reviewer: ```suggestion
`--kubelet-certificate-authority=<ca-string>`

```

Author: resolved

```

### 1.2.17 Ensure that the admission control plugin NodeRestriction is set (Automated)

**Result:** pass

**Remediation:**
Follow the Kubernetes documentation and configure NodeRestriction plug-in on kubelets.

> Then, edit the API server

Reviewer: ```suggestion
`--service-account-key-file=<filename>`

```

Converting to inline code since Docusaurus will throw an error because it treats `<filename>` as an HTML tag.

Author: resolved

> Then, edit the API server

Reviewer: ```suggestion
`--service-account-private-key-file=<filename>`

```

Converting to inline code since Docusaurus will throw an error because it treats `<filename>` as an HTML tag.

Author: resolved

> Then, edit the API server

Reviewer: ```suggestion
`--client-ca-file chmod 644 <filename>`

```

Converting to inline code since Docusaurus will throw an error because it treats `<filename>` as an HTML tag.

Author: resolved

> Then, edit the API server

Reviewer: ```suggestion
`chown root:root <filename>`

```

Converting to inline code since Docusaurus will throw an error because it treats `<filename>` as an HTML tag.

Author: resolved
